### PR TITLE
THREESCALE-10915: Fix Quick starts images in production mode via webpack

### PR DESCRIPTION
**Pull request by lvillen**
**What this PR does / why we need it**:

In production mode Quick starts images are not showing. This fix is related to #3729, which was a little confusing, and #3796, which proposes another way of fixing this problem. 

This PR fixes the broken images using webpack to handle them through the CopyWebpackPlugin. It also tidies up folders and unused images.

**Which issue(s) this PR fixes** 

[THREESCALE-10915](https://issues.redhat.com/browse/THREESCALE-10915)

**Verification steps** 
- configure porta for running in production mode
- precompile assets
- run Rails server in production mode
- verify that the images are visible in Quickstarts

### CODE REVIEW

File: app/assets/images/quick_starts/learn-about-accounts.png
Comment (jlledom) [inline app/assets/images/quick_starts/learn-about-accounts.png:1]: why renaming `quickstarts` to `quick_starts`?

**Comment by mayorova** [comment]
> However, if you could make Webpack add a fingerprint for images, then this Webpack approach would be better IMO because it would invalidate browser caches in case we update the images.

I have been looking into this hack:
https://github.com/3scale/porta/blob/master/app/javascript/src/QuickStarts/utils/replaceLinksExtension.ts#L15
that uses some Rails stuff: https://github.com/3scale/porta/blob/master/app/presenters/quickstarts_presenter.rb#L15


And I thought that we could probably follow a similar approach - the image links could be just replaced by the actual fingerprinted URLs, but I can't figure out how to get a fingerprinted URL even from Rails, even in production mode :cry:

File: app/assets/images/quick_starts/learn-about-accounts.png
Comment (josemigallas) [reply app/assets/images/quick_starts/learn-about-accounts.png:1]: I would leave the renaming for a different PR

**Comment by mayorova** [comment]
> However, if you could make Webpack add a fingerprint for images, then this Webpack approach would be better IMO because it would invalidate browser caches in case we update the images.

OK, so here is another one :sweat_smile:  https://github.com/3scale/porta/pull/3804

The digested images were already being added to `public/assets/quickstarts` folder, it's just that JavaScript doesn't know how to find them. So, I added some hack for that... 
Honestly, I am not a fan of it :upside_down_face: But I don't know whether there is an easier way to use digested images in JS :shrug: 

Otherwise, I'd just prefer https://github.com/3scale/porta/pull/3796 as I don't know what benefits would copying the images with the CopyWebpackPlugin have compared to just having the images already in `public/assets` :shrug:

**Comment by jlledom** [comment]
> > However, if you could make Webpack add a fingerprint for images, then this Webpack approach would be better IMO because it would invalidate browser caches in case we update the images.
> 
> OK, so here is another one 😅 #3804
> 
> The digested images were already being added to `public/assets/quickstarts` folder, it's just that JavaScript doesn't know how to find them. So, I added some hack for that... Honestly, I am not a fan of it 🙃 But I don't know whether there is an easier way to use digested images in JS 🤷
> 
> Otherwise, I'd just prefer #3796 as I don't know what benefits would copying the images with the CopyWebpackPlugin have compared to just having the images already in `public/assets` 🤷

I've been taking a look at this, and I think the best approach is #3796.

Is possible to make webpacker generate the images with the digest, I made it locally, you only need to include the images from somewhere, that is either from JS/TS:

```javascript
import * as image from 'path/to/image.png'
```
or from a view template:
```ruby
image_pack_tag 'path/to/image.png'
```
However the only place we link the images from is the `yaml` files, so in order to make it work with webpacker we would need to convert the `yaml` files to some kind of template that is processed to call one of the methods above. Options:
  1. Use Rails templates, that is, create a `slim` template from `yaml`, call `image_pack_tag 'pat/to/image.png'` in the template, and create a ruby service that renders the view and returns the proper `yaml`.
  2. Use some templating from Webpack itself, which must exist, maybe can be done with React, I know very little about this.

This seems a lot of work IMO, I think it's not worth it, and that's why I suggest going with #3796, since #3804 seems to hacky for me.

**Comment by jlledom** [comment]
Also. I think this is unused:

https://github.com/3scale/porta/blob/c3b094856e864aa942bcb54663c08d6475fa96f0/config/webpack/environment.js#L37-L44

I don't see any `JSON` being created from those `yaml`. Nor in the filesystem or the `manifest.json`

**Pull request closed without merging**